modules/pipeline/pipengine/actionexecutor/plugins/scheduler/logic/events.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/reference"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const (
+	DefaultEventsLimit = 5
+)
+
+var metadataAccessor = meta.NewAccessor()
+
+// SearchEvents finds events about the specified object.
+func SearchEvents(client corev1client.EventsGetter, objOrRef runtime.Object) (*corev1.EventList, error) {
```

Comment:
how long will the method cost?

Reply:
it`s similar to kubectl describe, handles the continue parameter returned by the API server when using list options